recipe/meta.yaml

```diff
@@ -213,33 +220,80 @@ outputs:
       doc_url: https://docs.nvidia.com/cuda/cuquantum/latest/cutensornet/
       dev_url: https://github.com/NVIDIA/cuQuantum
+  - name: custatevec
+    version: 1.7.0
+    build:
+      number: {{ sdk_build }}
+      string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ mpi_suffix }}"
+      script:
+        - "${RECIPE_DIR}/custatevec/cusv-install"
+      run_exports:
+        - {{ pin_subpackage("custatevec", max_pin="x") }} {{ mpi_prefix }}_*
+      ignore_run_exports:
+        - {{ compiler("cuda") }}
+      missing_dso_whitelist:
+        - '*/libcublas.so*'    # [cuda_compiler_version in ("11.8",)]
+        - '*/libcublasLt.so*'  # [cuda_compiler_version in ("11.8",)]
+      post-link: post-link
+    requirements:
+      build:
+        - {{ stdlib('c') }}
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - {{ compiler("cuda") }}
+      host:
+        - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version in ("11.8", "12.0", "12.6")]
+        - libcublas                                 # [cuda_compiler_version in ("12.0", "12.6")]
+      run:
+        - cuda-version >={{ cuda_major }}.0,<{{ cuda_major + 1 }}      # [cuda_compiler_version in ("11.8", "12.0", "12.6")]
+        - {{ pin_compatible("libcublas", max_pin="x", min_pin="x") }}  # [cuda_compiler_version in ("12.0", "12.6")]
+        - cudatoolkit  # [cuda_compiler_version == "11.8"]
+      run_constrained:
+        - openmpi <0.a0  # [mpi == "nompi"]
+        - mpich   <0.a0  # [mpi == "nompi"]
+    test:
+      requires:
+        - git
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - {{ compiler("cuda") }}
+        - cuda-version {{ cuda_compiler_version }}.*
+      files:
+        - ./common/*
+        - ./custatevec/*
+      commands:
+        - ./common/detect-glibc
+        - ./common/validate-glibc
+        - ./custatevec/cusv-tests {{ sdk_version_string }}
+    about:
+      home: https://developer.nvidia.com/cuquantum-sdk
+      license: LicenseRef-cuQuantum-Software-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/cuquantum/latest/license.html
+      license_file: LICENSE
+      summary: 'cuStateVec: A High-Performance Library for State Vector Quantum Simulators'
+      description: |
+        NVIDIA cuStateVec is a high-performance GPU library dedicated to operations with state vectors
+        for expressing quantum algorithms. cuStateVec is a component of the NVIDIA cuQuantum SDK.
 
+        The packages cuquantum, custatevec, and cutensornet are governed by the NVIDIA cuQuantum
+        Software License Agreement (EULA). By downloading and using the packages,
+        you accept the terms and conditions of the NVIDIA cuQuantum EULA -
+        https://docs.nvidia.com/cuda/cuquantum/license.html
+      doc_url: https://docs.nvidia.com/cuda/cuquantum/latest/custatevec/
+      dev_url: https://github.com/NVIDIA/cuQuantum
   - name: cuquantum-python
```

Comment:
Because the top-level `requirements.host` doesn't include python, conda-smithy isn't recognizing this feedstock as a python feedstock, so cuquantum-python is only being built with python=3.12 (the latest version). However, if we add python to the top-level requirements, the build matrix becomes 36 or 48 builds!? ((3 or 4) python * 2 CUDA * 2 archs * 3 mpi)

My understanding is that most of the build matrix would be redundant because cuquantum python is indifferent to mpi variant? It might be possible to prune the build matrix using skip statements? Otherwise, if the python builds are easily separable from the C-API, I suggest reactivating the cuquantum-python-feedstock.

2 cuda * 2 archs * 3 mpi = 12
2 cuda * 2 archs * 3 python = 16

If cuquantum-python is actually indifferent to the CUDA major version, also, then 

1 cuda * 2 archs * 3 python = 6

Since in a quick look, it doesn't seem tht cuquantum-python actually uses nvcc or links to the CTK directly.

cc: @leofang

Reply:
I was talking to Leo, and it seems that in older releases, we were able to trick conda-smithy/conda-build into building all python variants in one pipeline. For example, there would be 12 pipelines instead of 36 and each pipeline would build all three python variants (or skip all python variants). However, that doesn't seem to be working anymore.

It may be possible to replicate this behavior by manually adding the python variants to conda_build_config like how the mpi variants are manually added. However, I don't recommend this approach because it is non-standard and will likely not be compatible with channel-wide migrations of python packages.

Therefore, I think the best approach is to re-split the python package back into its own feedstock. Trading the convenience of having one feedstock for reduced complexity in the recipe.

Reply:
I removed the `cuquantum-python` output from this recipe. I'll take care of updating the recipe for cuquantum-python feedstock once it's unarchived and I am added as maintainer.